crates/primitives/starknet/src/transaction/mod.rs

```diff
@@ -304,6 +313,35 @@ impl Transaction {
         // However it also means we need to copy/paste internal code from the tx.execute() method.
     }
 
+    /// Verifies transaction nonce
+    ///
+    /// # Arguments
+    ///
+    /// * `self` - The transaction to verify the nonce for
+    /// * `account_tx_context` - The transaction context for the account
+    /// * `state` - The state to verify the nonce on
+    ///
+    /// # Returns
+    ///
+    /// * `TransactionExecutionResult<()>` - The result of the nonce verification
+    pub fn verify_nonce(
+        &self,
+        account_tx_context: &AccountTransactionContext,
+        state: &mut dyn State,
+    ) -> TransactionExecutionResult<()> {
+        if account_tx_context.version == TransactionVersion(StarkFelt::from(0)) {
+            return Ok(());
+        }
+        let currrent_nonce = state.get_nonce_at(account_tx_context.sender_address)?;
+        if currrent_nonce != account_tx_context.nonce {
+            return Err(TransactionExecutionError::InvalidNonce {
+                expected_nonce: account_tx_context.nonce,
+                actual_nonce: currrent_nonce,
+            });
+        }
+
+        Ok(state.increment_nonce(account_tx_context.sender_address)?)
```

Comment:
this is probably not needed as already done in `apply_state_diffs` in `pallets/starknet/lib.rs`